File: benchmarks/xlang/src/sieve.cpp
#include <cstdio>
#include <chrono>
#include <vector>
int main() {
    const long n = 10000000;
    auto t0 = std::chrono::steady_clock::now();
    std::vector<unsigned char> composite(n + 1, 0);
    for (long i = 2; i * i <= n; i++)
        if (!composite[i])
            for (long j = i * i; j <= n; j += i) composite[j] = 1;
    long count = 0;
    for (long p = 2; p <= n; p++) if (!composite[p]) count++;
    auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(
        std::chrono::steady_clock::now() - t0).count();
    printf("CHECK %ld\nMS %lld\n", count, (long long)ms);
}

File: benchmarks/xlang/src/sieve.go
package main
import ("fmt"; "time")
func main() {
    const n = 10000000
    t0 := time.Now()
    composite := make([]byte, n+1)
    for i := 2; i*i <= n; i++ {
        if composite[i] == 0 {
            for j := i * i; j <= n; j += i { composite[j] = 1 }
        }
    }
    count := 0
    for p := 2; p <= n; p++ {
        if composite[p] == 0 { count++ }
    }
    fmt.Printf("CHECK %d\nMS %d\n", count, time.Since(t0).Milliseconds())
}
